### PR TITLE
Move cudaDeviceSchedule* to device init

### PR DESCRIPTION
**Pull request by wraythex**

### CODE REVIEW

File: cudawrapper.c [hunk 1]
```diff
@@ -372,14 +372,23 @@ gpu_ecm (mpz_t f, mpz_t x, int param, mpz_t firstsigma, mpz_t n, mpz_t go,
...
       goto end_gpu_ecm;
-  
+
+  int schedule = 0;
+  if (use_cgbn)
+    {
+#ifdef HAVE_CGBN_H
+    schedule = 1;
+#else 
+    outputf (OUTPUT_ERROR, "cgbn not included, using yield schedule");
```
Comment (sethtroisi) [inline cudawrapper.c]: This will error on line 482.

Maybe

int schedule = (use_cgbn & HAVE_CGBN_H) ? 1 : 0

File: cudakernel.cu [hunk 2]
```diff
@@ -68,7 +68,7 @@ getNumberOfBlockPerMultiProcessor (int major, int minor)
 
 extern "C" 
 int 
-select_and_init_GPU (int device, unsigned int *number_of_curves, int verbose)
+select_and_init_GPU (int device, unsigned int *number_of_curves, int verbose, int schedule)
```
Comment (sethtroisi) [inline cudakernel.cu:71]: Thanks, this was on my TODO list!

File: cudawrapper.c (hunk 1 above)
Comment (wraythex) [reply cudawrapper.c]: I've left the error on line 482 alone.  If line 482 happens, then the program exits.  This is expected in your original code, right?  I think the only thing we need to change is to remove the "#else" here.  No need to let the user know about this issue if the program will quit at line 482 later.

Also, your proposed solution would probably fail to compile if HAVE_CGBN_H isn't defined.

I'll remove the "#else".

Comment (sethtroisi) [reply cudawrapper.c]: how about 

`int schedule = use_cgbn ? 0 : 1`

technically if you call ecm with `-gpu -cgbn` and cgbn wasn't configured the card will be initialized with the wrong scheduler but it's going to exit 0.001 seconds later.

Comment (wraythex) [reply cudawrapper.c]: Sure, that sounds good.  It looks cleaner too.  I'll update that.

Comment (wraythex) [reply cudawrapper.c]: I've updated that line.  You can see the commit down below.

**Comment by sethtroisi** [comment]
Thanks, It's nice to see other people trying out the code and contributing back :+1:

**Pull request merged**